projects/angular/src/data/datagrid/providers/items.ts

```diff
@@ -24,8 +26,15 @@ export class Items<T = any> {
 
   /**
    * Tracking function to identify objects. Default is reference equality.
+   *
+   * @deprecated in v15 and scheduled for removal in v17
+   */
+  iteratorTrackBy: TrackByFunction<T> = (_index, item) => item;
+
+  /**
+   * New tracking function to identify objects. If provided, this will be used instead of `iteratorTrackBy`.
    */
-  trackBy: TrackByFunction<T> = (_index, item) => item;
+  datagridTrackBy: ClrDatagridItemsTrackByFunction<T>;
 
```

Comment:
Adding a trackBy getter here would reduce all the if/else logic in selection.js, and make this change easier to remove in the future. 

```
get trackBy() { 
  return this.datagridTrackBy || this.iteratorTrackBy;
}
```

Reply:
Unfortunately, that won't work because these functions do not have compatible signatures.

Reply:
Gotcha. I still think a proxy function here would prevent us from having to add the ternary and if/else logic throughout selection.js and would simplify removing it later. 

```
  get trackBy() {
    return (_index, item) => this.datagridTrackBy ? this.datagridTrackBy(item) : this.iteratorTrackBy(_index, item);
  }
```

Reply:
That won't work. I need to be able to call `datagridTrackBy` without the index.

Reply:
I can try conditionally looking up the index, but the code would still have conditionally and potentially be more confusing because it would have to conditionally pass `undefined` as the index without a clear reason why at the callsite.

Reply:
I'll leave it up to you. The current instance is duplicative and can lead to confusion. There are a lot of ways to proxy this and make it easier to implement and remove later.

Reply:
I wrote a proxy for this for this. There is a precondition that must be met before the proxy `trackBy` function is used (`items.datagridTrackBy` or `items.all` must be set), but the code isn't as confusing as I thought it would be.